### PR TITLE
Jest-based snapshot testing (first steps)

### PR DESCRIPTION
**Pull request by czosel**
This shows a _very basic_ integration of jest-based snapshot testing. I hope it can be a starting point for future work in this direction. I wanted to share it early because unfortunately I don't have much free time to work on this right now.

```bash
npm run jest #run snapshot tests
npm run jest -- -u #update snapshots
```

refs #152

### CODE REVIEW

File: test/snapshot/array.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  it('of arrays', () => {
+    expect(
+      parser.parseEval(
+        [
+          'array(',
+          '  array("item1", "item2"),',
+          '  array("item3", "item4"),',
+          '  array("item5", "item6")',
+          ')',
+        ].join('\n'),
```
[the remark this answers is not in the document]
Comment (b4dnewz) [reply test/snapshot/array.test.js]: awesome, very readable. it should be easy now to refactor all tests

[the remark this answers is not in the document]
Comment (czosel) [reply test/snapshot/array.test.js]: Hmm, I guess using ES6 template literals instead of `join` also makes this quite readable.

**Comment by czosel** [comment]
@ichiriac Feel free to check out the PR and play around with it a bit. If you have the feeling that this is going in the right direction, we could continue migrating tests, and run the two test suites side-by-side for a while. After that, you can make the call on which of the old tests still need to stay and which can go :smiley:

**Comment by ichiriac** [comment]
many thanks @czosel, it's really awesome !

**Pull request merged**